shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -227,9 +227,43 @@ - (BOOL)application:(UIApplication*)application
     continueUserActivity:(NSUserActivity*)userActivity
       restorationHandler:(void (^)(NSArray* __nullable restorableObjects))restorationHandler {
 #endif
-  return [_lifeCycleDelegate application:application
-                    continueUserActivity:userActivity
-                      restorationHandler:restorationHandler];
+  if ([_lifeCycleDelegate application:application
+                 continueUserActivity:userActivity
+                   restorationHandler:restorationHandler]) {
+    return YES;
+  } else if (userActivity.activityType == NSUserActivityTypeBrowsingWeb) {
+    return NO;
+  } else {
+    NSURLComponents* components = [NSURLComponents componentsWithURL:userActivity.webpageURL
```

Comment:
Can you get the query and path off the `userActivity.webpageURL` URL if it's nonnull instead of creating a `NSURLComponents`?